lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplGroupByAccTest.java

```diff
@@ -0,0 +1,264 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable;
+import pro.taskana.TaskanaConfiguration;
+import pro.taskana.classification.api.ClassificationService;
+import pro.taskana.classification.api.models.ClassificationSummary;
+import pro.taskana.common.api.BaseQuery.SortDirection;
+import pro.taskana.common.api.security.CurrentUserContext;
+import pro.taskana.task.api.TaskService;
+import pro.taskana.task.api.models.ObjectReference;
+import pro.taskana.task.api.models.TaskSummary;
+import pro.taskana.testapi.TaskanaConfigurationModifier;
+import pro.taskana.testapi.TaskanaInject;
+import pro.taskana.testapi.TaskanaIntegrationTest;
+import pro.taskana.testapi.builder.ObjectReferenceBuilder;
+import pro.taskana.testapi.builder.TaskAttachmentBuilder;
+import pro.taskana.testapi.builder.TaskBuilder;
+import pro.taskana.testapi.builder.WorkbasketAccessItemBuilder;
+import pro.taskana.testapi.security.WithAccessId;
+import pro.taskana.workbasket.api.WorkbasketPermission;
+import pro.taskana.workbasket.api.WorkbasketService;
+import pro.taskana.workbasket.api.models.WorkbasketSummary;
+
+@TaskanaIntegrationTest
+@DisabledIfEnvironmentVariable(named = "DB", matches = "ORACLE")
+public class TaskQueryImplGroupByAccTest implements TaskanaConfigurationModifier {
+  @TaskanaInject TaskService taskService;
+  @TaskanaInject WorkbasketService workbasketService;
+  @TaskanaInject CurrentUserContext currentUserContext;
+  @TaskanaInject ClassificationService classificationService;
+
+  ClassificationSummary defaultClassificationSummary;
+  WorkbasketSummary defaultWorkbasket;
+  TaskSummary taskSummary1;
+  TaskSummary taskSummary2;
+  TaskSummary taskSummary3;
+
+  @Override
+  public TaskanaConfiguration.Builder modify(TaskanaConfiguration.Builder builder) {
+    return builder.useSpecificDb2Taskquery(false);
+  }
+
+  @WithAccessId(user = "user-1-1")
+  @BeforeAll
+  void setup() throws Exception {
+    defaultClassificationSummary =
+        defaultTestClassification().buildAndStoreAsSummary(classificationService, "businessadmin");
+    defaultWorkbasket = createWorkbasketWithPermission();
+    ObjectReference sor2 =
+        ObjectReferenceBuilder.newObjectReference()
+            .company("FirstCompany")
+            .value("FirstValue")
+            .type("SecondType")
+            .build();
+    ObjectReference por2 = defaultTestObjectReference().build();
+    taskSummary2 =
+        taskInWorkbasket(defaultWorkbasket)
+            .primaryObjRef(por2)
+            .objectReferences(sor2)
+            .due(Instant.parse("2022-11-10T09:45:00.000Z"))
+            .name("Name2")
+            .attachments(
+                TaskAttachmentBuilder.newAttachment()
+                    .channel("A")
+                    .classificationSummary(defaultClassificationSummary)
+                    .objectReference(por2)
+                    .build())
+            .buildAndStoreAsSummary(taskService);
```

Comment:
It's because of the due date, as Tasks are sorted by due date in this test by default. It was important to add Tasks in a different order than their due dates to check if they really get sorted.  So the 1, 2, 3 numbers refer to the order that the Tasks will have after they have been sorted. I'm not sure if it's better the other way round.